clarity/src/vm/functions/arithmetic.rs

```diff
@@ -503,14 +503,12 @@ pub fn native_bitwise_left_shift(input: Value, pos: Value) -> InterpreterResult<
         match input {
             Value::Int(input) => {
                 let result = input
-                    .checked_shl(u32_val)
-                    .ok_or(RuntimeErrorType::ArithmeticOverflow)?;
+                    .wrapping_shl(u32_val);
```

Comment:
Since you're already committing to masking the `shamt` value by the VM word size, why not just do that here?

```rust
let shamt = u32::try_from(u128_val & 0xffffffff).expect("FATAL: lower 32 bits did not convert to u32);
/* ... */
let result = input.wrapping_shl(shamt);
```

Reply:
Then, this code path never fails.

Reply:
Updated to use your implementation.  Tests and docs updated.